src/index.js

```diff
@@ -168,38 +170,41 @@ class StickyTable extends React.Component {
       ['leftColumn', 'width', 'leftStickyColumnCount'],
       ['rightColumn', 'width', 'rightStickyColumnCount']
     ].forEach(([stickyKey, sizeKey, countPropKey]) => {
-      var insets = [0];
-      var count = props[countPropKey];
-      var netInset = 0;
+      var insets = [];
 
-      //HINT we only want this loop for the second sticky and up
-      for (s = 1; s < count; s++) {
-        var node = stickyKey === 'header' || stickyKey === 'leftColumn' ? cellNodes[0] : cellNodes[cellNodes.length - 1];
+      if (props[countPropKey] > 1) {
+        insets = [0]
+        var count = props[countPropKey];
+        var netInset = 0;
 
-        if (node) {
-          var boundingRect = node.getBoundingClientRect();
+        // HINT we only want this loop for the second sticky and up
+        for (s = 1; s < count; s++) {
+          var node = stickyKey === 'header' || stickyKey === 'leftColumn' ? cellNodes[s - 1] : cellNodes[cellNodes.length - s];
 
-          netInset += boundingRect[sizeKey];
-        }
+          if (node) {
+            var boundingRect = node.getBoundingClientRect();
+            netInset += boundingRect[sizeKey];
+          }
 
-        insets.push(netInset);
+          insets.push(netInset);
+        }
       }
 
       stickyInsets[stickyKey] = insets;
     });
 
     //HINT avoid a render unless there's actually a change
     if (JSON.stringify(stickyInsets) !== JSON.stringify(this.state.stickyInsets)) {
-      this.setState({stickyInsets});
+      this.setState({ stickyInsets });
```

Comment:
lots of opinionated styles mixed in with a functional pull request - I'm personally not a fan of the spacing style changes made in this PR, but do appreciate the functional code

Reply:
I can revert them if you want - I think I have a global styling that somehow overwrote the `.eslint` or I used the format on save from vscode; not sure